docs/modules/release-notes/pages/0.27.adoc

```diff
@@ -19,30 +21,329 @@ To get started, follow xref:pkl-cli:index.adoc#installation[Installation].#
+
+=== `const local` object members
+
+The `const` modifier can be applied to object members, provided that they are also `local` (https://github.com/apple/pkl/pull/678[#678]).
+
+Currently, regular object members are not allowed to have the `const` modifier. This introduces an artificial pain point. 
+
+=== Pkl CLI Changes
+
+==== New CLI Flags
+
+The following common flag has been added:
+
+`--color`::
+Format messages with ANSI color codes.
+
+==== New command: `pkl analyze imports`
+
+As part of the set of APIs added for <<import-analysis,import analysis>>, a new subcommand has been added called `analyze imports` (https://github.com/apple/pkl/pull/695[#695]).
+
+For details, consult the xref:pkl-cli:index.adoc#command-analyze-imports[CLI documentation].
+
+=== Kotlin/Java code generator improvements
+
+=== Pkl Gradle Plugin Changes
+
+==== New `AnalyzeImportsTask`
+
+A new task, `AnalyzeImportsTask`, is introduced (https://github.com/apple/pkl/pull/695[#695]).
+
+This task is the Gradle analogy to `pkl analyze imports`.
+
+For more details, consult the xref:pkl-gradle:index.adoc#analyze-imports[documentation].
+
+==== Tracked file outputs
+
+Tasks created by the Pkl plugin now declare tracked output files (https://github.com/apple/pkl/pull/403[#403]).
+
+This means that downstream tasks do not need to declare an explicit dependency on the Pkl task.
+
+For example, assuming that `evalPkl` is an `EvalTask`:
+
+.build.gradle.kts
+[source,diff]
+----
+ val myGradleTask by tasks.registering {
+   inputs.files(evalPkl)
+-  dependsOn(evalPkl) // <1>
+ }
+----
+<1> No longer necessary to declare this dependency.
+
+[[transitive-modules-computed-by-default]]
+==== `transitiveModules` computed by default 
+
+The `transitiveModules` property of a task is now computed by building the import graph of the source modules (https://github.com/apple/pkl/pull/695[#695]).
+
+This means that Pkl-related tasks no longer need to declare their set of transitive modules, because the pkl-gradle plugin will compute this automatically.
+
+NOTE: This adds latency to each task. To opt out of this behavior, set `transitiveModules` explicitly.
+
+=== Standard Library Changes
+
+==== New Methods
+
+New methods have been added to the standard library (https://github.com/apple/pkl/pull/666[#666], https://github.com/apple/pkl/pull/683[#683]).
+
+
+|===
+|Method |Description
+
+|{uri-stdlib-String}#splitLimit()[`String.splitLimit()`]
+|Split a string around matches of a pattern at most N times.
+
+|Column 1, row 2
+|Column 2, row 2
+
+|Column 1, row 3
+|Column 2, row 3
+
+|Column 1, row 4
+|Column 2, row 4
+
+|Column 1, row 5
+|Column 2, row 5
+|===
+
+==== `String` to `Number` conversion improvements.
+
+The `String` to `Number` converter methods, for example,  {uri-stdlib-StringToInt}[`String.toInt()`], can now handle underscore separators (https://github.com/apple/pkl/pull/578[#578], https://github.com/apple/pkl/pull/580[#580]).
```

Comment:
You can't link at sentence granularity, but I think I'd put a reference in to the Language Reference mentioning underscores; https://github.com/holzensp/pkl/blob/91a4e490a11b50c69ee04214635393b51f2b1816/docs/modules/language-reference/pages/index.adoc#L114-L115

Reply:
Thought about this, but, wasn't really sure where to link. I think this sentence alone is clear enough, though, especially with the example right below it.